pkg/api/preflight.go

```diff
@@ -14,19 +14,8 @@ type PreflightRequest struct {
 
 // ValidationResult is the validation result to return in the deployment preflight response body
 type ValidationResult struct {
-	Status ValidationStatus            `json:"status"`
-	Error  *ManagementErrorWithDetails `json:"error,omitempty"`
-}
-
-type ManagementErrorWithDetails struct {
-	// Code - The error code returned from the server.
-	Code *string `json:"code,omitempty"`
-	// Message - The error message returned from the server.
-	Message *string `json:"message,omitempty"`
-	// Target - The target of the error.
-	Target *string `json:"target,omitempty"`
-	// Details - Validation error.
-	Details *[]ManagementErrorWithDetails `json:"details,omitempty"`
+	Status ValidationStatus `json:"status"`
+	Error  *CloudErrorBody  `json:"error,omitempty"`
```

Comment:
This was more from the older PR, where we discussed making a ValidationResult instead of adding to the existing CloudErrorBody, and just use inheritance. The format of CloudErrorBody is the standard but might be called a different name.

Reply:
@mociarain All errors in Azure are dictated to use the OData error format, but in this specific API they expect an additional field. 

http://docs.oasis-open.org/odata/odata-json-format/v4.0/os/odata-json-format-v4.0-os.html#_Toc372793091